src/util/config.cpp

```diff
@@ -1145,6 +1146,46 @@ void configt::set_from_symbol_table(
   // lib, string_abstraction not stored in namespace
 }
 
+/// Sets the number of bits used for object addresses
+/// \param cmdline The command line
+/// \param symbol_table The symbol table
+void configt::set_addr_bits(
+  const cmdlinet &cmdline, const symbol_tablet &symbol_table)
+{
+  bv_encoding.addr_bits=bv_encoding.default_addr_bits;
+
+  if(cmdline.isset("addr-bits"))
+  {
+    bv_encoding.addr_bits=
+      unsafe_string2unsigned(cmdline.get_value("addr-bits"));
+    if(!(0<bv_encoding.addr_bits &&
+         bv_encoding.addr_bits<ansi_c.pointer_width))
+    {
+      throw "addr-bits must be positive and less than the pointer width ("+
+        std::to_string(ansi_c.pointer_width)+") ";
+    }
+
+    return;
+  }
+
+  // set addr_bits according to entry point language
+  if(symbol_table.has_symbol(CPROVER_PREFIX "_start"))
```

Comment:
Use `else if` and avoid duplicate error messages by just using the "DATA_INVARIANT" below?